zookeeper/core/component.py

```diff
@@ -331,8 +346,17 @@ def __component_str__(instance):
                 value = utils.missing
             else:
                 raise e from None
+        parent_instance = next(
+            utils.generate_component_ancestors_with_field(instance, field_name), None
+        )
+        if parent_instance is not None:
+            is_inherited = base_getattr(parent_instance, field_name) == value  # type: ignore
+        else:
+            is_inherited = False
         field_strings.append(
-            _field_key_val_str(field_name, value, color=True, single_line=False)
+            _field_key_val_str(
+                field_name, value, is_inherited, color=True, single_line=False
+            )
```

Comment:
Is there any difference between `__component_str__` and `__component_repr__`? I don't see any, so then one could just call the other?

Reply:
Yeah, it's the difference between `repr(instance)` and `str(instance)`. The former is just a single line, the latter prints the nested stuff.

Reply:
Since the code is identical other than `color=True, single_line=False`, do you think it would be worth moving it to a new function that just takes those two as arguments and is then called from both the current functions?

Reply:
Nice, yeah I've done that now and it makes it all a bunch shorter and cleaner